exchange/messaging.go

```diff
@@ -385,7 +386,14 @@ func HasKeys() bool {
 var privFileName = "privateMessagingKey.pem"
 var pubFileName = "publicMessagingKey.pem"
 
+var KeyLock *sync.Mutex
```

Comment:
Both threads could call GetKeys at the same time and both find KeyLock to be nil, then both allocate a new lock and use it, one overwrites the other and you end up with the racing keys problem anyway.

Reply:
fixed